### ISSUE TITLE
SegFault for some getindex calls

### ISSUE THREAD

**Issue by denizyuret** [JuliaGPU/CuArrays.jl#38]
Please see https://github.com/denizyuret/Knet.jl/issues/198.

**Comment by maleadt** [comment]
The `warning: ignoring debug info with an invalid version (0) in ` error is because you're running from `release-0.6`, which isn't supported, but shouldn't break anything. I don't have time to look into it right now, but you could building LLVM with assertions enabled.

**Comment by denizyuret** [comment]
Is there any 0.6 that is supported?  How do I build LLVM with assertions
enabled?

**Comment by ilkerkesen** [comment]
That snippet gives me CUDA illegal memory access error,

https://gist.github.com/ilkerkesen/c921d41894b8667d8acc6e71803f04fe

**Comment by denizyuret** [comment]
I think the answer to my second question is: https://llvm.org/docs/CMake.html

**Comment by vchuravy** [comment]
Do build LLVM with assertions enabled you create `Make.user` file with

```
FORCE_ASSERTIONS=1
LLVM_ASSERTIONS=1
```

And the you purge you need to reconfigure your LLVM build or just do a `make -C deps distclean-llvm`

0.6.0 should work fine.

**Comment by maleadt** [comment]
0.6.0 is supported, 0.6.1 can be buggy. 0.6.2 will be supported, but `release-0.6` currently isn't (for version check reasons that do make sense). 0.7 is also supported, but requires a different version of LLVM.jl (tagged, so it should install automatically).


@denizyuret your code triggers an assertion in a Julia pass:

```
julia> @show summary(a[:,[2,4],2])
julia: julia-0.6/src/llvm-gcroot.cpp:796: void (anonymous namespace)::JuliaGCAllocator::allocate_frame(): Assertion `bb != NULL' failed.
```

I'll try to investigate next week. Might be caused by CUDAnative generating invalid IR (possibly only invalid with regard to the design of Julia's codegen and passes, undocumented requirements etc).

**Comment by MikeInnes** [comment]
But these don't come up in GPUArrays.jl's implementation, right? So just removing the CuArrays kernel should fix this?

**Comment by denizyuret** [comment]
I can confirm that as @SimonDanisch suggested commenting out https://github.com/JuliaGPU/CuArrays.jl/blob/master/src/CuArrays.jl#L13, i.e. include("indexing.jl") stopped the error and now the test runs fine.

**Comment by MikeInnes** [comment]
Closed with 00276e3c234188d728a94c12d88674839de642dd, see also https://github.com/JuliaGPU/GPUArrays.jl/issues/90.
